src/taipy/core/notification/event.py

```diff
@@ -71,30 +74,60 @@ class Event:
-        self.creation_date = datetime.now()
-        self.entity_id = entity_id
-        self.entity_type = entity_type
-        self.operation = self.__preprocess_operation(operation, entity_type)
-        self.attribute_name = self.__preprocess_attribute_name(attribute_name, operation)
-
-    @classmethod
-    def __preprocess_attribute_name(cls, attribute_name: Optional[str], operation: EventOperation) -> Optional[str]:
-        if operation in _NO_ATTRIBUTE_NAME_OPERATIONS and attribute_name is not None:
-            raise InvalidEventAttributeName
-        return attribute_name
+    entity_type: EventEntityType
+    operation: EventOperation
+    entity_id: Optional[str] = None
+    config_id: Optional[str] = None
+    attribute_name: Optional[str] = None
+    attribute_value: Optional[Any] = None
+
+    metadata: dict = field(default_factory=dict)
+    creation_date: datetime = field(init=False)
+
+    def __post_init__(self):
+        # Creation date
+        super().__setattr__("creation_date", datetime.now())
 
-    @classmethod
-    def __preprocess_operation(cls, operation: EventOperation, entity_type: EventEntityType) -> EventOperation:
-        if entity_type in _UNSUBMITTABLE_ENTITY_TYPES and operation == EventOperation.SUBMISSION:
+        # Check operation:
+        if self.entity_type in _UNSUBMITTABLE_ENTITY_TYPES and self.operation == EventOperation.SUBMISSION:
             raise InvalidEventOperation
-        return operation
+
+        # Check attribute name:
+        if self.operation in _NO_ATTRIBUTE_NAME_OPERATIONS and self.attribute_name is not None:
+            raise InvalidEventAttributeName
+
+
+@singledispatch
+def make_event(
+    entity: Any,
+    operation: EventOperation,
+    /,
+    attribute_name: Optional[str] = None,
+    attribute_value: Optional[Any] = None,
+    **kwargs,
+) -> Event:
+    """Helper function to make an event for this entity with the given `EventOperation^` type.
+    In case of `EventOperation.UPDATE^` events, an attribute name and value must be given.
+
+    Parameters:
+        entity (Any): The entity object to generate an event for.
+        operation (EventOperation^): The operation of the event. The possible values are:
+            <ul>
+                <li>CREATION</li>
+                <li>UPDATE</li>
+                <li>DELETION</li>
+                <li>SUBMISSION</li>
+            </ul>
+        attribute_name (Optional[str]): The name of the updated attribute for a `EventOperation.UPDATE`.
+            This argument is always given in case of an UPDATE.
+        attribute_value (Optional[Any]): The value of the udated attribute for a `EventOperation.UPDATE`.
+            This argument is always given in case of an UPDATE.
+        **kwargs (dict[str, any]): Any extra information that would be passed to the metadata event.
+            Note: you should pass only simple types: str, float, double as values."""
+    raise Exception(f"Unexpected entity type: {type(entity)}")
```

Comment:
*Notes to reviewers*:
This function must be overloaded for each entity supporting event creation. This one would be called if the interpreted does not find any for a particular entity type.